lab_2/main.py

```diff
@@ -108,7 +269,41 @@ def accumulate_diff_stats(original_text_tokens: tuple, suspicious_text_tokens: t
      'sentence_lcs_length': list,
      'difference_indexes': list}
     """
-    pass
+    checks = [
+        isinstance(original_text_tokens, tuple),
+        isinstance(suspicious_text_tokens, tuple),
+    ]
+
+    if not all(checks):
+        return {}
+
+    origin = original_text_tokens
+    suspicious = suspicious_text_tokens
+
+    text_plagiarism = calculate_text_plagiarism_score(origin, suspicious)
+    sentence_plagiarism = []
+    sentence_lcs_length = []
+    difference_indexes = []
+
+    for sent_1, sent_2 in zip(origin, suspicious):
+        lcs_length = find_lcs_length(sent_1, sent_2, plagiarism_threshold)
+        lcs_matrix = fill_lcs_matrix(sent_1, sent_2)
+        lcs = find_lcs(sent_1, sent_2, lcs_matrix)
+        plagiarism_score = calculate_plagiarism_score(lcs_length, sent_2)
+        diffs = find_diff_in_sentence(origin, suspicious, lcs)
+
+        sentence_plagiarism.append(plagiarism_score)
+        sentence_lcs_length.append(lcs_length)
+        difference_indexes.append(diffs)
+
+    diff_stats = {
```

Comment:
can you work directly with this dictionary?

Reply:
Yep, fixed it.